### ISSUE TITLE
Make parametrized tests xdist-compatible

### ISSUE THREAD

**Issue by abalakh** [SatelliteQE/airgun#4]
<a href="https://github.com/abalakh"><img src="https://avatars0.githubusercontent.com/u/11719030?v=4" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [abalakh](https://github.com/abalakh)**
_Monday Feb 19, 2018 at 12:34 GMT_
_Originally opened as https://github.com/abalakh/airgun/issues/22_

----

(robottelo issue)
Currently test runs with parametrized tests using xdist are failing with msgs like:
```
collecting 0 items / 3 errors
==================================== ERRORS ====================================
_____________________________ ERROR collecting gw1 _____________________________
Different tests were collected between gw0 and gw1. The difference is:
```
This is due to random data generated inside `parametrize`. We should probably use the same hash seed across different workers. More info here: https://github.com/pytest-dev/pytest/issues/1075

**Comment by abalakh** [comment]
Apart from using the same hash seed we can stop sending random data in test names - instead of `test_positive_create[dfh7yt94hf9]` introduce names for datafactory values, so tests will always be named like `test_positive_create[alphanumeric]`, `test_positive_create[utf-8]` etc. That's even better naming and it will also avoid encountering such issue.

**Comment by omaciel** [comment]
I like your suggestion @abalakh and it does make it easier to see what type of data was passed (the actual value can always be gotten by looking at logs, if needed)

**Comment by abalakh** [comment]
Addressed in SatelliteQE/robottelo#5904 , you can see updated names in the test results there:
```python
[...]
tests/foreman/ui_airgun/test_activationkey.py::test_positive_create_with_envs[alphanumeric] PASSED [ 22%]
tests/foreman/ui_airgun/test_architecture.py::test_positive_create[alphanumeric] PASSED [ 36%]
tests/foreman/ui_airgun/test_architecture.py::test_positive_delete[alpha] PASSED [ 45%]
tests/foreman/ui_airgun/test_operatingsystem.py::test_positive_create[numeric] PASSED [ 68%]
[...]
```
